### DISCUSSION TITLE
Display coverage from a subdir with Vitest UI

### DISCUSSION THREAD

**Discussion by andrewjtorres** [vitest-dev/vitest#3916] [Q&A] [answered]
Hi Vitest Team!

Is it possible to have Vitest UI display code coverage results from a sub-directory?

To elaborate, I have the following configuration:
```typescript
const config = defineConfig(({ command }) => ({
  test: {
    include: ['src/**/?(*.)unit.test.[jt]s?(x)'],
    environment: 'happy-dom',
    watch: false,
    root: rootDir,
    reporters: process.env.CI ? 'dot' : undefined,
    setupFiles: ['config/vitest/setup.ts'],
    coverage: {
      include: ['src/**/*.[jt]s?(x)'],
      exclude: ['src/**/?(*.).test.[jt]s?(x)', 'src/mocks/**', 'src/utils/spec.tsx', 'src/config.ts'],
      all: true,
      reportsDirectory: 'test-reports',
      reporter: [
        [process.env.CI ? 'cobertura' : 'text'],
        [
          'html',
          {
            subdir: 'coverage',
          },
        ],
      ],
    },
    mockReset: true,
    resolveSnapshotPath: (path: string, extension: string) => `${path}${extension}`,
  },
}))
```

When I run `npx vitest --coverage --ui --watch` it appears as if the coverage is being picked up, but when I select the coverage icon in the top left, the following error appears in the console:

![Screenshot 2023-08-08 at 9 16 15 AM](https://github.com/vitest-dev/vitest/assets/450495/dce2a115-4f82-48ef-92a6-b2373f85af46)

Based on the error message, the `reportsDirectory` option is being respected, but the `subdir` option associated with the html coverage reporter is not. Are there any configuration options that can be provided to Vitest UI to allow for this setup?

Thank you all for your time and the amazing work you are doing. 🙂

**Comment by userquin** [comment]
Try running `npx vitest --coverage.enabled=true --ui --watch` or adding `coverage.enabled: true` to your configuration, it should work

**Reply by userquin** [reply]
You should also add `html` to reporters: not required, should work with your configuration, the docs seems outdated

**Reply by andrewjtorres** [reply]
Running `npx vitest --coverage.enabled=true --ui --watch` results in the same error appearing in the console. When I run the original command (`npx vitest --coverage --ui --watch`), the coverage is being generated properly, but Vitest UI doesn't appear to pickup on the nested location. Based on the config I provided above, I expect Vitest UI to look for the coverage `index.html` file at `http://localhost:51204/test-reports/coverage/index.html` as opposed to `http://localhost:51204/test-reports/index.html`.

**Reply by userquin** [reply]
ok, we should also include/check the `subdir` entry and update how we check/access the reporters entry, I'll try to send a PR to update docs and fix the problem

**Reply by andrewjtorres** [reply]
That is very much appreciated. Thank you for the quick response and timely action plan. 🙂

**Reply by userquin** [reply]
ok, working on my local with `subdir: 'coverage'`, updating docs to include `coverage.enabled`:

![imagen](https://github.com/vitest-dev/vitest/assets/6311119/a6d49d15-d8a9-4964-ba6b-d8162adaa86c)

**Reply by userquin** [reply, marked as answer]
@andrewjtorres https://github.com/vitest-dev/vitest/pull/3917

**Reply by andrewjtorres** [reply]
Awesome! 🙌🏼

**Reply by melroy89** [reply]
I actually have a similar issue, but here the whole Vitest UI is hosted within a subdirectory using GitLab pages..

For example: `https://someproject.pages.domain.com/someproject` in that case the coverage file is still in the sub-directory relative to the project.. But since the project is under the `/someproject` URL, it still can't find the coverage directory.. Because the VItest UI websites goes now to `domain.com/coverage` instead of `domain.com/someproject/coverage`.. 

Maybe a simple HTML base tag (in the `<head>` section) could maybe already help? Changing the `subdir: 'myproject/coverage'` feels wrong, since the same output works fine locally when running the `vitest --ui` command for example or a simple local python server: `python3 -m http.server 8000`

Its just that the Vitest UI can't handle if its hosted in a sub directory very well.

**Reply by hi-ogawa** [reply]
@melroy89 Hosting entire html output and coverage under sub directory should become possible out-of-the-box in https://github.com/vitest-dev/vitest/pull/9626.
